### PR TITLE
Test ingest pdf

### PR DESCRIPTION
**Pull request by helenCNode**
Some edits that were made necessary by small issues encountered during ingestion of PDFs into the DB.

### CODE REVIEW

File: retrieval/db/ingest.py [hunk 1]
```diff
@@ -283,91 +307,144 @@ def webpages_to_ingested_data(
...
-                        web_metadata = row["web_metadata"]
-                        if type(web_metadata) is list:
-                            web_metadata = web_metadata[0]
-                        metadata = pdf.guess_metadata(date_guess=web_metadata.get("publishDate"), indent="\t")
-                        metadata["location"] = link
-                        doc = LangchainDocument(page_content=text, metadata=metadata)
-                        docs.append(doc)
+            #   print("\n\n", button_links_doc_titles, "\n\n")
 
+            desirable_file_name_and_link_tuples = [
```
Comment (beingkk) [inline retrieval/db/ingest.py]: Would be great to add a comment on what you mean by "desirable file name"

File: retrieval/db/ingest.py [hunk 2]
```diff
@@ -283,91 +307,144 @@ def webpages_to_ingested_data(
...
+        for file_name, link, title_guess in desirable_file_name_and_link_tuples:
+
+            if not doc_already_in_db(link):
+                if file_name:
+                    path = PDF_PATH / file_name
                 else:
-                    logger.info(f"\nPDF {file_name} did not seem to have any text – ignoring")
+                    path = link
+
+                if cautious:
```
Comment (beingkk) [inline retrieval/db/ingest.py]: Adding a comment on what "cautious" mean would be great

**Review by beingkk — APPROVED** [review]
Thank you @helenCNode, looks good to merge!

File: retrieval/db/ingest.py (hunk 1 above)
Comment (helenCNode) [reply retrieval/db/ingest.py]: Added

File: retrieval/db/ingest.py (hunk 2 above)
Comment (helenCNode) [reply retrieval/db/ingest.py]: Added

**Pull request merged**